fetch-node.js

```diff
@@ -6,7 +6,7 @@ function wrapFetchForNode(fetch) {
   // Support schemaless URIs on the server for parity with the browser.
   // https://github.com/matthew-andrews/isomorphic-fetch/pull/10
   return function (u, options) {
-    if (u instanceof fetch.Request) {
+    if (fetch && fetch.Request && u instanceof fetch.Request) {
```

Comment:
Does this mean that, when using a cookie, schemaless URIs will not work?

Reply:
If I do not add this check then while running a sample request in node.js returns the following error
I don't think this is related to schema-less uris. This check would protect us from cases where fetch is undefined.
```
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.myFetch (D:\sdk\ms-rest-js\node_modules\fetch-ponyfill\fetch-node.js:9:43)
    at Object.<anonymous> (D:\sdk\ms-rest-js\dist\lib\util\utils.js:299:33)
    at Generator.next (<anonymous>)
    at D:\sdk\ms-rest-js\dist\lib\util\utils.js:9:71
    at Promise (<anonymous>)
    at __awaiter (D:\sdk\ms-rest-js\dist\lib\util\utils.js:5:12)
    at Object.dispatchRequest (D:\sdk\ms-rest-js\dist\lib\util\utils.js:262:12)
    at RequestPipeline.requestSink (D:\sdk\ms-rest-js\dist\lib\requestPipeline.js:47:22)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

This is how my request object looks like:
```
{ 
  url: 'https://management.azure.com/subscriptions/subId/providers/Microsoft.Storage/storageAccounts?api-version=2015-06-15',
  method: 'GET'
}
```
After having this check the request passes successfully.

Reply:
I guess any edge case this introduces is going to be very rare to encounter, so it's fine. I don't like reassigning a function parameter though. Would you mind doing something along the line of:

```javascript
function wrapFetchForNode(nodeFetch, useCookie) {
  const fetch = useCookie ? fetchCookie(nodeFetch) : nodeFetch;

  // Support schemaless URIs on the server for parity with the browser.
  // https://github.com/matthew-andrews/isomorphic-fetch/pull/10
  return function (u, options) {
    if (u instanceof nodeFetch.Request) {
      return fetch(u, options);
    }

    var url = u.slice(0, 2) === '//' ? 'https:' + u : u;

    return fetch(url, options);
  };
}
```